pkg/services/authn/clients/oauth.go

```diff
@@ -204,6 +210,29 @@ func (c *OAuth) RedirectURL(ctx context.Context, r *authn.Request) (*authn.Redir
 	}, nil
 }
 
+func (c *OAuth) Logout(ctx context.Context, user identity.Requester, info *login.UserAuth) (*authn.Redirect, bool) {
+	token := c.oauthService.GetCurrentOAuthToken(ctx, user)
+
+	if err := c.oauthService.InvalidateOAuthTokens(ctx, info); err != nil {
+		namespace, id := user.GetNamespacedID()
+		c.log.FromContext(ctx).Error("Failed to invalidate tokens", "namespace", namespace, "id", id, "error", err)
+	}
+
+	redirctURL := getOAuthSignoutRedirectURL(c.cfg, c.oauthCfg)
+	if redirctURL == "" {
+		c.log.FromContext(ctx).Debug("No signout redirect url configured")
+		return nil, false
+	}
+
+	if isOICDLogout(redirctURL) && token != nil && token.Valid() {
+		if idToken, ok := token.Extra("id_token").(string); ok {
+			redirctURL = withIDTokenHint(redirctURL, idToken)
```

Comment:
Having done exactly that a few days ago, I think you sadly need to do it in two steps if `token.Extra("id_token")` is `nil` then that panics. Then IMO, it's safe to assume that if it's set then it's a string.
```suggestion
		if idToken := token.Extra("id_token"); idToken != nil {
			redirctURL = withIDTokenHint(redirctURL, idToken.(string))
```

Reply:
I was wrong: 
```go
func TestService_Token(t *testing.T) {
	token := &oauth2.Token{
		AccessToken:  "aa.bb.cc",
		Expiry:       time.Now(),
		RefreshToken: "bb.cc.dd",
		TokenType:    "blabla",
	}

	_, ok := token.Extra("id_token").(string)

	assert.False(t, ok)
}
```